src/ripple/json/json_writer.h

```diff
@@ -175,6 +176,138 @@ std::string valueToQuotedString ( const char* value );
 /// \see Json::operator>>()
 std::ostream& operator<< ( std::ostream&, const Value& root );
 
+//------------------------------------------------------------------------------
+
+// Helpers for stream
+namespace detail {
+
+template <class Write>
+void
+write_string(Write&& write, std::string const& s)
```

Comment:
I don't see a reason to use a forwarding ref here (we never move write). I'd use a `const&` here.

Reply:
Good point.  It has become habit to do `&&` for callbacks, but clearly not needed here.

Reply:
I don't think there's one right answer. The forwarding ref handles non-const callbacks and `const&` obviously does not. In generic code like this should we really be prohibiting non-const callbacks?

I do prefer `const&` here, but you're closer to the problem, so you'll know better.

Reply:
Probably not worth prohibiting in generic code, but the use cases here are limited/well defined, so I'm ok leaving with the more restrictive version for now.